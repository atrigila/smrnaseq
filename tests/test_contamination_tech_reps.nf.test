
nextflow_pipeline {

    name "Test Workflow main.nf - test_contamination_tech_reps"
    script "main.nf"
    profile "test_contamination_tech_reps"
    tag "test_contamination_tech_reps"
    tag "pipeline"

    test("test_contamination_tech_reps") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("software_versions") },
                { assert workflow.trace.succeeded().size() == 73 },

                { assert snapshot(
                    path("$outputDir/mirna_quant/reference/mature.fa_igenome.fa"),
                    path("$outputDir/mirna_quant/reference/hairpin.fa_igenome.fa"),
                    path("$outputDir/mirna_quant/reference/mature.fa_igenome.fa_idx.fa"),
                    path("$outputDir/mirna_quant/reference/hairpin.fa_igenome.fa_idx.fa")
                ).match("mirna_quant_reference") },

                { assert snapshot(
                    path("$outputDir/mirna_quant/seqcluster/final/Clone1_N3_trimmed.filtered_trimmed.fastq.gz").exists(),
                    path("$outputDir/mirna_quant/seqcluster/final/Clone1_N1_trimmed.filtered_trimmed.fastq.gz").exists(),
                    path("$outputDir/mirna_quant/seqcluster/final/Clone1_N1.filtered_trimmed.fastq.gz").exists()
                ).match("mirna_quant_seqcluster") },

                { assert snapshot(
                    path("$outputDir/mirna_quant/bam/mature/Clone1_N1_trimmed_mature.sorted.bam").exists(),
                    path("$outputDir/mirna_quant/bam/mature/Clone1_N1_trimmed_mature.sorted.bam.csi").exists(),
                    path("$outputDir/mirna_quant/bam/mature/Clone1_N3_trimmed_mature.sorted.bam").exists(),
                    path("$outputDir/mirna_quant/bam/mature/Clone1_N3_trimmed_mature.sorted.bam.csi").exists(),
                    path("$outputDir/mirna_quant/bam/mature/Clone1_N1_mature.sorted.bam").exists(),
                    path("$outputDir/mirna_quant/bam/mature/Clone1_N1_mature.sorted.bam.csi").exists(),
                    path("$outputDir/mirna_quant/bam/mature/Clone1_N1_trimmed_mature.sorted.idxstats"),
                    path("$outputDir/mirna_quant/bam/mature/Clone1_N1_trimmed_mature.sorted.stats"),
                    path("$outputDir/mirna_quant/bam/mature/Clone1_N3_trimmed_mature.sorted.idxstats"),
                    path("$outputDir/mirna_quant/bam/mature/Clone1_N1_trimmed_mature.sorted.flagstat"),
                    path("$outputDir/mirna_quant/bam/mature/Clone1_N1_mature.sorted.idxstats"),
                    path("$outputDir/mirna_quant/bam/mature/Clone1_N3_trimmed_mature.sorted.flagstat"),
                    path("$outputDir/mirna_quant/bam/mature/Clone1_N3_trimmed_mature.sorted.stats"),
                    path("$outputDir/mirna_quant/bam/mature/Clone1_N1_mature.sorted.stats"),
                    path("$outputDir/mirna_quant/bam/mature/Clone1_N1_mature.sorted.flagstat"),
                    path("$outputDir/mirna_quant/bam/hairpin/Clone1_N1_trimmed_mature_hairpin.sorted.bam").exists(),
                    path("$outputDir/mirna_quant/bam/hairpin/Clone1_N1_trimmed_mature_hairpin.sorted.bam.csi").exists(),
                    path("$outputDir/mirna_quant/bam/hairpin/Clone1_N3_trimmed_mature_hairpin.sorted.bam").exists(),
                    path("$outputDir/mirna_quant/bam/hairpin/Clone1_N3_trimmed_mature_hairpin.sorted.bam.csi").exists(),
                    path("$outputDir/mirna_quant/bam/hairpin/Clone1_N1_mature_hairpin.sorted.bam").exists(),
                    path("$outputDir/mirna_quant/bam/hairpin/Clone1_N1_mature_hairpin.sorted.bam.csi").exists(),
                    path("$outputDir/mirna_quant/bam/hairpin/Clone1_N3_trimmed_mature_hairpin.sorted.idxstats"),
                    path("$outputDir/mirna_quant/bam/hairpin/Clone1_N1_trimmed_mature_hairpin.sorted.idxstats"),
                    path("$outputDir/mirna_quant/bam/hairpin/Clone1_N1_trimmed_mature_hairpin.sorted.stats"),
                    path("$outputDir/mirna_quant/bam/hairpin/Clone1_N1_trimmed_mature_hairpin.sorted.flagstat"),
                    path("$outputDir/mirna_quant/bam/hairpin/Clone1_N1_mature_hairpin.sorted.flagstat"),
                    path("$outputDir/mirna_quant/bam/hairpin/Clone1_N1_mature_hairpin.sorted.idxstats"),
                    path("$outputDir/mirna_quant/bam/hairpin/Clone1_N3_trimmed_mature_hairpin.sorted.flagstat"),
                    path("$outputDir/mirna_quant/bam/hairpin/Clone1_N3_trimmed_mature_hairpin.sorted.stats"),
                    path("$outputDir/mirna_quant/bam/hairpin/Clone1_N1_mature_hairpin.sorted.stats")
                ).match("mirna_quant_bam") },

                { assert snapshot(
                    path("$outputDir/mirna_quant/mirtop/mirtop.tsv").exists(),
                    path("$outputDir/mirna_quant/mirtop/mirtop_rawData.tsv").exists(),
                    path("$outputDir/mirna_quant/mirtop/mirtop.gff").exists(),
                    path("$outputDir/mirna_quant/mirtop/mirna.tsv").exists(),
                    path("$outputDir/mirna_quant/mirtop/stats/mirtop_stats.txt").exists()
                ).match("mirna_quant_mirtop") },

                { assert snapshot(
                    path("$outputDir/mirna_quant/edger_qc/hairpin_counts.csv").exists(),
                    path("$outputDir/mirna_quant/edger_qc/hairpin_log2CPM_sample_distances.txt").exists(),
                    path("$outputDir/mirna_quant/edger_qc/hairpin_edgeR_MDS_distance_matrix.txt").exists(),
                    path("$outputDir/mirna_quant/edger_qc/hairpin_edgeR_MDS_plot_coordinates.txt").exists(),
                    path("$outputDir/mirna_quant/edger_qc/hairpin_logtpm.csv").exists(),
                    path("$outputDir/mirna_quant/edger_qc/hairpin_unmapped_read_counts.txt").exists(),
                    path("$outputDir/mirna_quant/edger_qc/mature_edgeR_MDS_distance_matrix.txt").exists(),
                    path("$outputDir/mirna_quant/edger_qc/hairpin_normalized_CPM.txt").exists(),
                    path("$outputDir/mirna_quant/edger_qc/mature_edgeR_MDS_plot_coordinates.txt").exists(),
                    path("$outputDir/mirna_quant/edger_qc/mature_counts.csv").exists(),
                    path("$outputDir/mirna_quant/edger_qc/hairpin_logtpm.txt").exists(),
                    path("$outputDir/mirna_quant/edger_qc/mature_logtpm.csv").exists(),
                    path("$outputDir/mirna_quant/edger_qc/mature_log2CPM_sample_distances.txt").exists(),
                    path("$outputDir/mirna_quant/edger_qc/mature_normalized_CPM.txt").exists(),
                    path("$outputDir/mirna_quant/edger_qc/mature_logtpm.txt").exists(),
                    path("$outputDir/mirna_quant/edger_qc/mature_unmapped_read_counts.txt").exists()
                ).match("mirna_quant_edger_qc") },

                { assert snapshot(
                    path("$outputDir/contaminant_filter/index/fasta_bidx.3.bt2"),
                    path("$outputDir/contaminant_filter/index/fasta_bidx.2.bt2"),
                    path("$outputDir/contaminant_filter/index/fasta_bidx.4.bt2"),
                    path("$outputDir/contaminant_filter/index/fasta_bidx.rev.2.bt2"),
                    path("$outputDir/contaminant_filter/index/fasta_bidx.1.bt2"),
                    path("$outputDir/contaminant_filter/index/fasta_bidx.rev.1.bt2")
                ).match("contaminant_filter_index") },

                { assert snapshot(
                    path("$outputDir/contaminant_filter/blat/filtered.fa")
                ).match("contaminant_filter_blat") },

                { assert snapshot(
                    path("$outputDir/contaminant_filter/map/Clone1_N3_trimmed.tRNA.filter.unmapped.contaminant.fastq").exists(),
                    path("$outputDir/contaminant_filter/map/filtered.Clone1_N3_trimmed_tRNA.stats"),
                    path("$outputDir/contaminant_filter/map/filtered.Clone1_N1_trimmed_tRNA.stats"),
                    path("$outputDir/contaminant_filter/map/Clone1_N1_trimmed.tRNA.filter.unmapped.contaminant.fastq").exists(),
                    path("$outputDir/contaminant_filter/map/filtered.Clone1_N1_tRNA.stats"),
                    path("$outputDir/contaminant_filter/map/Clone1_N1.tRNA.filter.unmapped.contaminant.fastq").exists(),
                    path("$outputDir/contaminant_filter/map/Clone1_N3_trimmed.cDNA.filter.unmapped.contaminant.fastq").exists(),
                    path("$outputDir/contaminant_filter/map/filtered.Clone1_N3_trimmed_cDNA.stats"),
                    path("$outputDir/contaminant_filter/map/Clone1_N1_trimmed.cDNA.filter.unmapped.contaminant.fastq").exists(),
                    path("$outputDir/contaminant_filter/map/filtered.Clone1_N1_trimmed_cDNA.stats"),
                    path("$outputDir/contaminant_filter/map/Clone1_N1.cDNA.filter.unmapped.contaminant.fastq").exists(),
                    path("$outputDir/contaminant_filter/map/filtered.Clone1_N1_cDNA.stats"),
                    path("$outputDir/contaminant_filter/map/Clone1_N3_trimmed.ncRNA.filter.unmapped.contaminant.fastq").exists(),
                    path("$outputDir/contaminant_filter/map/filtered.Clone1_N3_trimmed_ncRNA.stats"),
                    path("$outputDir/contaminant_filter/map/Clone1_N3_trimmed.filter.contaminant.sam").exists(),
                    path("$outputDir/contaminant_filter/map/filtered.Clone1_N1_trimmed_ncRNA.stats"),
                    path("$outputDir/contaminant_filter/map/Clone1_N1_trimmed.ncRNA.filter.unmapped.contaminant.fastq").exists(),
                    path("$outputDir/contaminant_filter/map/Clone1_N1_trimmed.filter.contaminant.sam").exists(),
                    path("$outputDir/contaminant_filter/map/filtered.Clone1_N1_ncRNA.stats"),
                    path("$outputDir/contaminant_filter/map/Clone1_N1.ncRNA.filter.unmapped.contaminant.fastq").exists(),
                    path("$outputDir/contaminant_filter/map/Clone1_N1.filter.contaminant.sam").exists()
                ).match("contaminant_filter_map") },

                { assert snapshot(
                    path("$outputDir/contaminant_filter/filter/Clone1_N1_trimmed.contamination_mqc.yaml"),
                    path("$outputDir/contaminant_filter/filter/Clone1_N1_trimmed.filtered.fastq.gz").exists(),
                    path("$outputDir/contaminant_filter/filter/Clone1_N3_trimmed.contamination_mqc.yaml"),
                    path("$outputDir/contaminant_filter/filter/Clone1_N3_trimmed.filtered.fastq.gz").exists(),
                    path("$outputDir/contaminant_filter/filter/Clone1_N1.filtered.fastq.gz").exists(),
                    path("$outputDir/contaminant_filter/filter/Clone1_N1.contamination_mqc.yaml")
                ).match("contaminant_filter_filter") },

                { assert snapshot(
                    path("$outputDir/bowtie_index/mirna_mature/fasta_bidx.1.ebwt"),
                    path("$outputDir/bowtie_index/mirna_mature/fasta_bidx.2.ebwt"),
                    path("$outputDir/bowtie_index/mirna_mature/fasta_bidx.4.ebwt"),
                    path("$outputDir/bowtie_index/mirna_mature/fasta_bidx.3.ebwt"),
                    path("$outputDir/bowtie_index/mirna_mature/fasta_bidx.rev.2.ebwt"),
                    path("$outputDir/bowtie_index/mirna_mature/fasta_bidx.rev.1.ebwt")
                ).match("bowtie_index_mirna_mature") },

                { assert snapshot(
                    path("$outputDir/bowtie_index/mirna_hairpin/fasta_bidx.1.ebwt"),
                    path("$outputDir/bowtie_index/mirna_hairpin/fasta_bidx.3.ebwt"),
                    path("$outputDir/bowtie_index/mirna_hairpin/fasta_bidx.4.ebwt"),
                    path("$outputDir/bowtie_index/mirna_hairpin/fasta_bidx.rev.1.ebwt"),
                    path("$outputDir/bowtie_index/mirna_hairpin/fasta_bidx.rev.2.ebwt"),
                    path("$outputDir/bowtie_index/mirna_hairpin/fasta_bidx.2.ebwt")
                ).match("bowtie_index_mirna_hairpin") },

                { assert snapshot(
                    path("$outputDir/mirtrace/mirtrace-stats-contamination_detailed.tsv"),
                    path("$outputDir/mirtrace/mirtrace-stats-length.tsv"),
                    path("$outputDir/mirtrace/mirtrace-stats-phred.tsv"),
                    path("$outputDir/mirtrace/mirtrace-stats-mirna-complexity.tsv"),
                    path("$outputDir/mirtrace/mirtrace-stats-contamination_basic.tsv"),
                    path("$outputDir/mirtrace/mirtrace-stats-qcstatus.tsv"),
                    path("$outputDir/mirtrace/mirtrace-stats-rnatype.tsv")
                ).match("mirtrace") },

                { assert snapshot(
                    path("$outputDir/mirtrace/qc_passed_reads.all.collapsed/Clone1_N1_trimmed.fastp.fasta"),
                    path("$outputDir/mirtrace/qc_passed_reads.all.collapsed/Clone1_N3_trimmed.fastp.fasta"),
                    path("$outputDir/mirtrace/qc_passed_reads.all.collapsed/Clone1_N1.fastp.fasta")
                ).match("mirtrace_qc_passed_reads.all.collapsed") },

                { assert snapshot(
                    path("$outputDir/mirtrace/qc_passed_reads.rnatype_unknown.collapsed/Clone1_N1_trimmed.fastp.fasta"),
                    path("$outputDir/mirtrace/qc_passed_reads.rnatype_unknown.collapsed/Clone1_N3_trimmed.fastp.fasta"),
                    path("$outputDir/mirtrace/qc_passed_reads.rnatype_unknown.collapsed/Clone1_N1.fastp.fasta")
                ).match("mirtrace_qc_passed_reads.rnatype_unknown.collapsed") },

            )
        }

    }

}
