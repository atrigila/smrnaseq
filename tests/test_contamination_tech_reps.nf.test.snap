{
    "mirtrace": {
        "content": [
            true,
            "mirtrace-stats-contamination_detailed.tsv:md5,663a7a5e7ebfc012a278186a96179790",
            "mirtrace-stats-contamination_basic.tsv:md5,6e8d8bc93931fbbded2899f31aac3e3d",
            "mirtrace-stats-length.tsv:md5,f687c7ba38cb6a4c4f7178aff385522e",
            "mirtrace-stats-mirna-complexity.tsv:md5,4b16d7d2767f8d2db1053f9cc879d508",
            "mirtrace-stats-phred.tsv:md5,07d8b5416961ef3725f86089df8db132",
            "mirtrace-stats-qcstatus.tsv:md5,71ed8e813730d698418de2d94f266c7d",
            "mirtrace-stats-rnatype.tsv:md5,3e98f069e8bb7492b3c80ac77831ccf6",
            true,
            "mirtrace-stats-contamination_detailed.tsv:md5,0b189d549c002cbcc70e2f740436ac3c",
            "mirtrace-stats-contamination_basic.tsv:md5,2fa99aa67f1f835abebedbddcd84770e",
            "mirtrace-stats-length.tsv:md5,72d9194d5e101d5575b978082a3a38f3",
            "mirtrace-stats-mirna-complexity.tsv:md5,215c1b6fa0965e6bf141fd26fe537364",
            "mirtrace-stats-phred.tsv:md5,9f7af4ed3888a41c9a9ad8fa926c8d62",
            "mirtrace-stats-qcstatus.tsv:md5,73177136ba84b3dfca600005695a3331",
            "mirtrace-stats-rnatype.tsv:md5,f1f950519e20706b1ada7d7c953a3b40",
            true,
            "mirtrace-stats-contamination_detailed.tsv:md5,e760eac5053b74a2bd3fe3bee33d020f",
            "mirtrace-stats-contamination_basic.tsv:md5,5e848ae99f2f54b6a663816d78657512",
            "mirtrace-stats-length.tsv:md5,4dab1089f2bd2f6d49b4409591a4e945",
            "mirtrace-stats-mirna-complexity.tsv:md5,8fcbc04a6538f4a36d19bbf9294016b1",
            "mirtrace-stats-phred.tsv:md5,2c94697ff2a9d1ff8507fda4d62b19a3",
            "mirtrace-stats-qcstatus.tsv:md5,650819c1549894183ee0558db2be75fa",
            "mirtrace-stats-rnatype.tsv:md5,ba98a0a2b7a3dd0be7ec667d9a2ffd91"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-06T13:46:55.643429949"
    },
    "software_versions": {
        "content": [
            "{BLAT_CDNA={blat=36}, BLAT_NCRNA={blat=36}, BOWTIE_MAP_HAIRPIN={bowtie=1.3.1, samtools=1.14}, BOWTIE_MAP_MATURE={bowtie=1.3.1, samtools=1.14}, BOWTIE_MAP_SEQCLUSTER={bowtie=1.3.1, samtools=1.14}, CAT_FASTQ={cat=8.3}, FASTP={fastp=0.23.4}, FILTER_STATS={BusyBox=1.32.1}, FORMAT_HAIRPIN={fastx_toolkit=0.0.14}, FORMAT_MATURE={fastx_toolkit=0.0.14}, GAWK_CDNA={gawk=5.3.0}, GAWK_NCRNA={gawk=5.3.0}, INDEX_CDNA={bowtie2=2.4.5}, INDEX_HAIRPIN={bowtie=1.3.1}, INDEX_MATURE={bowtie=1.3.1}, INDEX_NCRNA={bowtie2=2.4.5}, INDEX_TRNA={bowtie2=2.4.5}, MAP_CDNA={bowtie2=2.4.5}, MAP_NCRNA={bowtie2=2.4.5}, MAP_TRNA={bowtie2=2.4.5}, MIRTOP_COUNTS={mirtop=0.4.25}, MIRTOP_EXPORT={mirtop=0.4.25}, MIRTOP_GFF={mirtop=0.4.25}, MIRTOP_STATS={mirtop=0.4.25}, MIRTRACE_QC={mirtrace=1.0.1}, PARSE_HAIRPIN={seqkit=2.6.1}, PARSE_MATURE={seqkit=2.6.1}, SAMTOOLS_FLAGSTAT={samtools=1.2}, SAMTOOLS_IDXSTATS={samtools=1.2}, SAMTOOLS_INDEX={samtools=1.2}, SAMTOOLS_SORT={samtools=1.2}, SAMTOOLS_STATS={samtools=1.2}, SEQCLUSTER_SEQUENCES={seqcluster=1.2.9}, SEQKIT_GREP_CDNA={seqkit=2.8.0}, SEQKIT_GREP_NCRNA={seqkit=2.8.0}, TABLE_MERGE={r-base=3.6.2}, Workflow={nf-core/smrnaseq=v2.3.2dev}}"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-13T22:05:38.72680003"
    },
    "mirna_quant_bam": {
        "content": [
            true,
            true,
            true,
            true,
            true,
            true,
            true,
            true,
            true,
            true,
            true,
            true,
            true,
            true,
            true,
            true,
            true,
            true
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-04T19:24:40.511510374"
    },
    "mirna_quant_edger_qc": {
        "content": [
            true,
            true,
            true,
            true,
            true,
            true,
            true,
            true,
            true,
            true,
            true,
            true,
            true,
            true,
            true,
            true
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-04T19:24:40.56322797"
    },
    "contaminant_filter_filter": {
        "content": [
            true,
            true,
            true
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-04T19:24:40.458015808"
    },
    "mirna_quant_mirtop": {
        "content": [
            true,
            true,
            true,
            true,
            true
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-04T19:24:40.610914296"
    }
}